workspace/TS100/Core/Threads/GUIThread.cpp

```diff
@@ -425,34 +426,86 @@ static void gui_solderingMode(uint8_t jumpToSleep) {
 	}
 	for (;;) {
 		ButtonState buttons = getButtonState();
-		switch (buttons) {
-		case BUTTON_NONE:
-			// stay
-			boostModeOn = false;
-			break;
-		case BUTTON_BOTH:
-			// exit
-			return;
-			break;
-		case BUTTON_B_LONG:
-			return;  // exit on back long hold
-			break;
-		case BUTTON_F_LONG:
-			// if boost mode is enabled turn it on
-			if (systemSettings.BoostTemp)
+		if (buttonsLocked && (systemSettings.lockingMode != 0)) { // If buttons locked
+			switch (buttons) {
+			case BUTTON_NONE:
+				// stay
+				boostModeOn = false;
+				break;
+			case BUTTON_BOTH_LONG:
+				// Unlock buttons
+				buttonsLocked = false;
+				OLED::setCursor(0, 0);
+				OLED::clearScreen();
+				OLED::setFont(0);
+				OLED::print(UnlockingKeysString);
+				OLED::refresh();
+				osDelay(1000);
```

Comment:
Could we use the wait for button press or timeout rather than a hard delay?

Reply:
Done!